app/org/sagebionetworks/bridge/dynamodb/DynamoTaskDao.java

```diff
@@ -68,15 +72,27 @@ public void setSchedulePlanService(SchedulePlanService schedulePlanService) {
     public void setUserConsentDao(UserConsentDao userConsentDao) {
         this.userConsentDao = userConsentDao;
     }
+    
+    @Autowired
+    public void setActivityService(ActivityService activityService) {
+        this.activityService = activityService;
+    }
 
     /** {@inheritDoc} */
     @Override
     public List<Task> getTasks(User user, DateTime endsOn) {
+        StudyIdentifier studyId = new StudyIdentifierImpl(user.getStudyKey());
+        
         Map<String,List<Task>> scheduledTasks = scheduleTasksForPlans(user, endsOn);
         List<Task> tasksToSave = Lists.newArrayList();
         for (String runKey : scheduledTasks.keySet()) {
             if (taskHasNotBeenSaved(user.getHealthCode(), runKey)) {
-                tasksToSave.addAll(scheduledTasks.get(runKey));
+                for (Task task : scheduledTasks.get(runKey)) {
```

Comment:
dao shouldn't rely on services. You'd need to pull this up to the service level, do the iteration there, then maybe call back to the dao to save a list of tasks that are verified not in the db.